### PR TITLE
KFSPTS-27197: IWant Document Show Full Page and Show Multiple Pages buttons added that allows a user to switch the presentation when edoc status is Initiated or Saved.

### PR DESCRIPTION
**Pull request by nkimble**

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/module/purap/document/IWantDocument.java [hunk 1]
```diff
@@ -50,6 +50,8 @@ public class IWantDocument extends FinancialSystemTransactionalDocumentBase impl
     private static final long serialVersionUID = 1L;
 
     private String step;
+    private String presentationMode;
+    private boolean usersRequestingPresentationModeChange;
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/purap/document/IWantDocument.java:54]: I see that these are both non-persisted fields, but I also see that you defined them on both the document itself and the document form. Can you please explain why you need them on both objects and not just on the form? (I see that the existing `step` variable is in both places, but do the other two variables also need to be on both?)

File: src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java [hunk 2]
```diff
@@ -37,7 +39,8 @@ public IWantDocumentForm() {
         newSourceLine = new IWantAccount();
         this.setDeptOrgKeyLabels(new ArrayList<KeyValue>());
         editingMode = new HashMap<Object,Object>();
-
+        presentationMode = CUPurapConstants.IWantPresentationModes.MULTIPLE_PAGE_MODE;
+        usersRequestingPresentationModeChange = true;
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java:43]: I'm only seeing `usersRequestingPresentationModeChange` being set to true. Is it supposed to lose its value and revert to false if the user takes a document action that doesn't call the related setter? If so, could the variable be renamed to make its intent clearer, or could this be implemented in a clearer fashion?

**Review by cah292 — CHANGES_REQUESTED** [review]
Changes look good overall, though I did have a few questions and suggestions.

File: src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java [hunk 3]
```diff
@@ -205,8 +211,38 @@ public List<ExtraButton> getExtraButtons() {
...
+                isMultiplePagesAllowed && isUsersRequestingPresentationModeChange()) {
+            extraButtons.add(createShowMultiplePagesButton());
+        }
+    }
+    
+    protected ExtraButton createShowFullPageButton() {
+        ExtraButton showFullPageButton = new ExtraButton();
+        showFullPageButton.setExtraButtonProperty("methodToCall.showFullPage");
+        showFullPageButton.setExtraButtonSource("${" + KFSConstants.EXTERNALIZABLE_IMAGES_URL_KEY + "}buttonsmall_continue.gif");
+        showFullPageButton.setExtraButtonAltText("Show Full Page");
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java:235]: This button name doesn't line up with the yellow-highlighted one that was mentioned in the new user story. Can you please update this if necessary?

The same goes for the other new button.

**Review by cah292 — CHANGES_REQUESTED** [review]
I noticed one other thing; sorry for not mentioning it in the main review.

File: src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java (hunk 2 above)
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentForm.java]: Thanks for asking about this boolean flag. It was used during initial development of this feature but subsequent logic adjustments have resulted in it no longer being needed. I have removed it.

**Review by nkimble — COMMENTED** [review]
Requested changes have been made. This is ready to be reviewed again.

File: src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentAction.java [hunk 4]
```diff
@@ -67,18 +67,35 @@ protected void loadDocument(KualiDocumentFormBase kualiDocumentFormBase) {
...
+                iWantForm.setStep(CUPurapConstants.IWantDocumentSteps.CUSTOMER_DATA_STEP);
+                iWantDocument.setStep(CUPurapConstants.IWantDocumentSteps.CUSTOMER_DATA_STEP);
+
+                if (StringUtils.isNotBlank(iWantDocument.getCurrentRouteToNetId())) {
+                    iWantForm.getNewAdHocRoutePerson().setId(iWantDocument.getCurrentRouteToNetId());
+                }
+
+            } else if (!iWantDocument.getDocumentHeader().getWorkflowDocument().isInitiated()) {
+              iWantForm.setStep(CUPurapConstants.IWantDocumentSteps.REGULAR);
+              iWantDocument.setStep(CUPurapConstants.IWantDocumentSteps.REGULAR);
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentAction.java:84]: There are a couple lines here where the indentation is still off a bit.

**Review by cah292 — CHANGES_REQUESTED** [review]
I noticed one very minor spot that would be good to clean up, but otherwise the revisions look good.

File: src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentAction.java (hunk 4 above)
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentAction.java]: fixed.

Comment (nkimble) [reply src/main/java/edu/cornell/kfs/module/purap/document/web/struts/IWantDocumentAction.java]: Formatting has been fixed and unit tests are running.

**Review by cah292 — COMMENTED** [review]
Changes look good, I'll merge this once the unit tests pass.

**Review by cah292 — APPROVED** [review]
Changes look good, unit tests pass and my concerns have been addressed. Merging.

**Pull request merged**